docs/configuration/http_conn_man/http_conn_man.rst

```diff
@@ -97,6 +100,44 @@ http_codec_options
   <config_cluster_manager_cluster_http_codec_options>` option. See the comment there about
   disabling HTTP/2 header compression.
 
+.. _config_http_conn_man_http2_settings:
+
+http2_settings
+  *(optional, object)* Additional HTTP/2 settings that are passed directly to the HTTP/2 codec.
+  Currently supported settings are:
+
+  hpack_table_size
+    *(optional, integer)* `Maximum Table Size`_ (in octets) that the encoder is permitted to use for
+    the dynamic HPACK table. Valid values range from 0 to 4294967295 (2^32 - 1) and defaults to 4096,
+    with 0 effectively disables header compression.
+
+  max_concurrent_streams
+    *(optional, integer)* `Maximum concurrent streams`_ allowed for peer on one HTTP/2 connection.
+    Valid values range from 0 to 2147483647 (2^31 - 1) and defaults to 2147483647.
+
+    NOTE: Total streams is 32-bit unsigned, one-side (client/server) is half that, and we also need to
+    exclude stream 0.
+
+.. _config_http_conn_man_http2_settings_initial_window_size:
+
+  initial_window_size
```

Comment:
Nit: can you rename this `initial_stream_window_size` now that we also have `initial_connection_window_size`?